### DISCUSSION TITLE
error 502 (Bad Gateway) on POST /upload

### DISCUSSION THREAD

**Discussion by hkovacs** [strapi/strapi#5985] [Legacy Topics]
**Describe the bug**
just updated to beta-20.1, tried uploading a file which i had been able to do successfully in previous version beta-20.

my fields are:
ref: "association" (my collection name)
refId: "5da979e69bd03850f90e35b3" (the association id)
field: "amendments" (the component name)
title: "test file upload" (self explanatory)
files: a-test-file.txt (the file that has been uploaded)

the POST does a save of the file and i can see the file in Media Library. however, i get a 502 error and strapi does a restart without listing any errors in logs.

**Steps to reproduce the behavior**

**Expected behavior**
I was getting success without error yesterday. I updated this morning and now i dont have success. i reviewed the git changes for 20.1 and dont see anything that might have caused this.

**Screenshots**
NA

**Code snippets**
NA

**System**

Node.js version: v10.16.3
NPM version: 6.9.0
Strapi version: Strapi v3.0.0-beta.20.1
Database: mongo v4.0.12
Operating system: ubuntu 18.04

**Additional context**
Thank you!

**Comment by derrickmehaffy** [comment]
Are you using Nginx? If so I'll convert this to a discussion as it isn't a bug with Strapi (502 Gateway errors do not relate to Strapi itself)

**Comment by hkovacs** [comment]
yes, nginx. ok, a discussion sounds good to me :)

any suggestions?

**Reply by derrickmehaffy** [reply]
Most likely nginx is having problems connecting to Strapi, generally due to a bad nginx config. Can you add your config here?

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Comment by hkovacs** [comment]
following https://strapi.io/documentation/3.0.0-beta.x/deployment/nginx-proxy.html solved it for me.

thanks @derrickmehaffy
